xbmc/platform/posix/filesystem/SMBFile.h

```diff
@@ -79,7 +79,7 @@ class CSMBFile : public IFile
   bool OpenForWrite(const CURL& url, bool bOverWrite = false) override;
   bool Delete(const CURL& url) override;
   bool Rename(const CURL& url, const CURL& urlnew) override;
-  int GetChunkSize() override { return 64*1024; }
+  int GetChunkSize() override { return 1; }
```

Comment:
This seems odd. I read the description in the OP but I still don't really understand this.

Reply:
they way I understand it: samba doesn't care about chunk size, so by returning 1 here, file cache is free to read as much data as possible from samba in one request.

Reply:
For reference, here's the `GetChunkSize()` doc comment (from IFile.h):
```
  /* Returns the minimum size that can be read from input stream.  *
   * For example cdrom access where access could be sector based.  *
   * This will cause file system to buffer read requests, to       *
   * to meet the requirement of CFile.                             *
   * It can also be used to indicate a file system is non buffered *
   * but accepts any read size, have it return the value 1         */
  virtual int  GetChunkSize() {return 0;}

```
I interpret that like it should return 1 for SMBFile, because it has no special requirements regarding read size. Before this commit however that would have caused FileCache to try to read one byte at a time. Well not really, because the now removed `CFile::DetermineChunkSize()` call in would have used the advancedsettings memorysize instead... 

64kiB was the maximum read request in SMB < 2.1 I believe, so possibly that is the origin of the old value, or maybe it was just a round number.

Reply:
@wsnipex Exactly!

In practice a 64kiB chunk size is also ok I believe, in my environment the reads have been a few MiBs each with this change so 64kiB just rounds that down slightly, maybe 1-5%

Reply:
I may have changed my mind a little bit. One thing that the old block size did was to ensure that FileCache doesn't do too *small* reads. Now it can do very small ones if the storage is fast. So maybe that value should be left alone after all.